lib/Runtime/Library/JavascriptLibrary.cpp

```diff
@@ -1054,29 +1054,29 @@ namespace Js
             scriptContext->ResetWeakReferenceDictionaryList();
             scriptContext->SetIsFinalized();
             scriptContext->GetThreadContext()->UnregisterScriptContext(scriptContext);
+            scriptContext->MarkForClose();
```

Comment:
not only for that, after this point, the recycler pointers hold by or can-be-reached-through scriptContext may be collected/invalid. most of those pointers was in Cache struct (which I moved to directly on JavascriptLibrary to simplify the cleanup path). I am using isClosed on scriptContext as barrier, and do those cleanup. if we do this at JavascriptLibrary::Dispose() time, it's too late because those recycler pointer might be collected already